code/obj/item/gun/kinetic.dm

```diff
@@ -11,6 +11,9 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 	var/ammo_cats = null
 	/// Does this gun have a special icon state for having no ammo lefT?
 	var/has_empty_state = FALSE
+	/// Does this gun have a special icon state it should flick to when fired?
+	var/has_fire_anim_state = FALSE
+	var/fire_anim_state = null
```

Comment:
These vars allows single-action guns to support firing animations. Previously uncocking the hammer would trigger the fire animation, since the fire animation is included in the base state icon.